### PR TITLE
New EnumerableChoiceField for autocompletion + pagination

### PR DESCRIPTION
**Pull request by joehybird**

### CODE REVIEW

File: creme/creme_core/tests/forms/test_enumerable.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+        enumerable = EnumerableChoiceSet(FakeContact._meta.get_field('user'), limit=limit)
+        choices, more = enumerable.choices()
+
+        self.assertListEqual(available_choices[:limit], [c.as_dict() for c in choices])
+        self.assertEqual(more, has_more)
+
+    def test_choices__selected(self):
+        enumerable = EnumerableChoiceSet(FakeContact._meta.get_field('sector'))
+        farming, industry, software = FakeSector.objects.order_by('pk').all()
```
Comment (genglert) [inline creme/creme_core/tests/forms/test_enumerable.py:101]: `.all()` is useless

File: creme/creme_core/tests/forms/test_enumerable.py [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+        self.assertEqual([
+            industry, sector_C
+        ], enumerable.to_python([
+            industry.pk, sector_C.pk
+        ]))
+
+    def test_to_python__invalid_value(self):
+        field = FakeContact._meta.get_field('sector')
+        enumerable = EnumerableChoiceSet(field)
+        self.assertEqual([], enumerable.to_python([99999]))
```
Comment (genglert) [inline creme/creme_core/tests/forms/test_enumerable.py:288]: `_CremeTestCase.UNUSED_PK` ??

File: creme/creme_core/tests/forms/test_enumerable.py [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+            (group, [c.as_dict() for c in choices]) for group, choices in field.choices
+        ])
+
+    def test_initial(self):
+        farming, industry, software = FakeSector.objects.order_by('pk').all()
+        field = EnumerableChoiceField(
+            FakeContact, 'sector', empty_label='No value...', initial=industry.pk
+        )
+
+        self.assertEqual(None, field.empty_label)
```
Comment (genglert) [inline creme/creme_core/tests/forms/test_enumerable.py:381]: assertIsNone

**Review by genglert — CHANGES_REQUESTED** [review]

File: CHANGELOG.txt [hunk 4]
```diff
@@ -19,6 +19,7 @@
     # The comboboxes with autocompletion now use Select2 tool and their visuals and behavior have slightly changed.
     # The old color picker 'gccolor' is replaced by the standard HTML5 color input.
     # The version of the shipped Closure compiler has been upgraded (2020-01-12 => 2022-09-05).
+    # Creme user selection fields now use paginated autocompletion.
```
[the remark this answers is not in the document]
Comment (joehybird) [reply CHANGELOG.txt:22]: yes for now only users

File: creme/creme_core/models/fields.py [hunk 5]
```diff
@@ -128,9 +130,23 @@ def deconstruct(self):
...
 
-        return super().formfield(**{'form_class': CremeUserChoiceField, **kwargs})
+        if self.get_tag(FieldTag.ENUMERABLE):
+            return Field.formfield(
+                self,
+                form_class=CremeUserEnumerableField,
+                model=self.model,
+                field_name=self.name,
+                required=not self.blank,
+                empty_label=_('Select a user...'),
```
[the remark this answers is not in the document]
Comment (joehybird) [reply creme/creme_core/models/fields.py:145]: Oh I forgot this one

File: creme/creme_core/tests/forms/test_enumerable.py (hunk 2 above)
Comment (joehybird) [reply creme/creme_core/tests/forms/test_enumerable.py:288]: Thx I wasn't able to retrieve it 😓

**Review by genglert — CHANGES_REQUESTED** [review]

**Review by genglert — APPROVED** [review]

**Pull request merged**